pkg/kudoctl/util/kudo/resources.go

```diff
@@ -0,0 +1,50 @@
+package kudo
+
+import (
+	"os"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/http"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/finder"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/repo"
+)
+
+// Resources tries to look for package files resolving the operator name to:
+// - a local tgz file
+// - a local directory
+// - a url to a tgz
+// - an operator name in the remote repository
+// in that order. Should there exist a local folder e.g. `cassandra` it will take precedence
+// over the remote repository package with the same name.
+// TODO: move this to the 'packages' package -- currently not possible because if a dependency cycle
```

Comment:
is this todo still relevant? I don't think so...

Reply:
Yes, very relevant. This function should be in `packages` but can't be there without further refactorings, see https://github.com/kudobuilder/kudo/pull/991#discussion_r338510126

Reply:
okay then I would probably not move it here, since we want to move it somewhere else anyway 🤷‍♀

Reply:
We should probably reorganize `packages` pgk similar to what we did for the `engine` pkg:
- have `packages/types.go` with all package types
- put everything else into `packages/*` sub-packages to avoid circular dependencies

This is the pattern that K8S uses a lot and it seems to make a lot of sense.